core/pkg/artifacts/downloader.go

```diff
@@ -86,6 +91,124 @@ func (ad *ArtifactDownloader) getArtifactManifest(artifactID string) (manifest M
 	return manifest, nil
 }
 
+// getBatchEntriesWithFileUrls fetches file urls for the entries in
+// entriesToFetch and returns a list of the formatted manifest entries to
+// download in the current batch.
+func (ad *ArtifactDownloader) getBatchEntriesWithFileUrls(
+	artifactID string,
+	entriesToFetch []gql.ArtifactManifestEntryInput,
+	manifest Manifest,
+	manifestEntries map[string]ManifestEntry,
+	nameToScheduledTime map[string]time.Time,
+) ([]ManifestEntry, error) {
+	var entries []ManifestEntry
+	now := time.Now()
+	response, err := gql.ArtifactFileURLsByManifestEntries(
+		ad.Ctx,
+		ad.GraphqlClient,
+		artifactID,
+		entriesToFetch,
+		manifest.StoragePolicyConfig.StorageLayout,
+		strconv.Itoa(int(manifest.Version)),
+		nil,
+	)
+	if err != nil {
+		return nil, err
```

Comment:
**suggestion:** should probably return the empty list that's already been declared
```suggestion
		return entries, err
```